### PR TITLE
Remove loader prop at data store level

### LINKED ISSUE #4911: Stop propagating Loader downward from Container Runtime
Related to #4910

Currently the ContainerRuntime propagates the Loader downward by way of its interface.  It shouldn't do this by default, and hosts that want to do so should provide it some other way, such as through the scope.  Existing downward consumers of the Loader will need to be fixed to use whatever the new pattern is.

### PR DESCRIPTION
**Pull request by heliocliu**
Fixes #4911

Loader prop at the data store level has been deprecated for many versions and bohemia has been updated to use the new path (through the container scope object instead e.g. `context.scope.ILoader`, which is set automatically when created by a Loader).  Other issues around the relative loader have already been addressed separately as well.

### CODE REVIEW

File: packages/loader/container-loader/src/loader.ts
Comment (anthony-murphy) [inline packages/loader/container-loader/src/loader.ts:81]: these changes seems breaking, and i'm also worried that the loader on the scope doesn't use the relative loader

Comment (ChumpChief) [inline packages/loader/container-loader/src/loader.ts:81]: Maybe safer to return a 404 response here rather than assert

Comment (heliocliu) [reply packages/loader/container-loader/src/loader.ts:81]: I'll change it to return the 404 response like Matt suggested?  In the old state where Boehmia was using the relative loader, all the requests pass through to the real loader anyway because they just weren't trying to resolve to the current container...  Imo this kind of goes back to a different issue I had with the loader object, where it's provided in a lot of places as the interface but the consumers *really* care about its implementation.  We could maybe provide an explicit RelativeLoader instead of an ILoader on the data store alongside the optional scope loader, but I'm not sure how useful that actually ends up being

File: packages/runtime/runtime-definitions/src/dataStoreContext.ts
Comment (vladsud) [inline packages/runtime/runtime-definitions/src/dataStoreContext.ts:260]: I think 0.37 here means you can't merge it right away, likely want to give it 2 releases to bake?

**Review by vladsud — APPROVED** [review]

File: packages/runtime/runtime-definitions/src/dataStoreContext.ts
Comment (heliocliu) [reply packages/runtime/runtime-definitions/src/dataStoreContext.ts:260]: yup will wait a while before merging

**Comment by anthony-murphy** [comment]
shouldn't this have a breaking note?

**Comment by heliocliu** [comment]
> shouldn't this have a breaking note?

@anthony-murphy it had a breaking note back when the deprecated tag was added.  should i add another one?  iirc previously i added them when something got deprecated rather than when it got removed, since the latter was sometimes causing problems at integration

**Comment by anthony-murphy** [comment]
I think we should do both. The first (depreciation) is more a notice, and the second is that actual break.

**Pull request merged**
